src/app/files.js

```diff
@@ -107,6 +107,47 @@ function Files (storage) {
     return files
   }
 
+  //
+  // Tree model for files
+  // {
+  //   'a': { }, // empty directory 'a'
+  //   'b': {
+  //     'c': {}, // empty directory 'b/c'
+  //     'd': { '/readonly': true, '/content': 'Hello World' } // files 'b/c/d'
+  //     'e': { '/readonly': false, '/path': 'b/c/d' } // symlink to 'b/c/d'
```

Comment:
> but in case of b/c, the content would be fs.b['/content'].c ... if c is a file, but fs.b.c if c is a directory.

I don't entirely follow. `/content` is only valid on a file which must be a leaf node.

> Where would you store "meta information" about directories?

When we get to that point, very likely I would store them with a different prefix in the storage API (e.g. `sol-meta:` and not `sol:`)